monai/transforms/spatial/array.py

```diff
@@ -596,34 +653,46 @@ def __call__(
                 )
             spatial_ornt = nib.orientations.ornt_transform(src, dst)
         new_affine = affine_ @ nib.orientations.inv_ornt_aff(spatial_ornt, spatial_shape)
-        _is_tensor = isinstance(data_array, torch.Tensor)
+
         spatial_ornt[:, 0] += 1  # skip channel dim
         spatial_ornt = np.concatenate([np.array([[0, 1]]), spatial_ornt])
         axes = [ax for ax, flip in enumerate(spatial_ornt[:, 1]) if flip == -1]
         if axes:
-            data_array = (
-                torch.flip(data_array, dims=axes) if _is_tensor else np.flip(data_array, axis=axes)  # type: ignore
-            )
+            data_array = torch.flip(data_array, dims=axes)
         full_transpose = np.arange(len(data_array.shape))
         full_transpose[: len(spatial_ornt)] = np.argsort(spatial_ornt[:, 0])
         if not np.all(full_transpose == np.arange(len(data_array.shape))):
-            if _is_tensor:
-                data_array = data_array.permute(full_transpose.tolist())  # type: ignore
-            else:
-                data_array = data_array.transpose(full_transpose)  # type: ignore
-        out, *_ = convert_to_dst_type(src=data_array, dst=data_array)
+            data_array = data_array.permute(full_transpose.tolist())
+
         new_affine = to_affine_nd(affine_np, new_affine)
-        new_affine, *_ = convert_to_dst_type(src=new_affine, dst=affine, dtype=torch.float32)
+        new_affine, *_ = convert_data_type(new_affine, torch.Tensor, dtype=torch.float32, device=data_array.device)
 
-        if self.image_only:
-            return out
-        return out, affine, new_affine
+        data_array = convert_to_tensor(data_array, track_meta=get_track_meta())
+        if get_track_meta():
+            self.update_meta(data_array, new_affine)
+            self.push_transform(data_array, extra_info={"old_affine": affine_np})
+        return data_array
 
+    def update_meta(self, img, new_affine):
+        img.affine = new_affine
 
-class Flip(Transform):
+    def inverse(self, data: torch.Tensor) -> torch.Tensor:
+        transform = self.pop_transform(data)
+        # Create inverse transform
+        orig_affine = transform[TraceKeys.EXTRA_INFO]["old_affine"]
+        orig_axcodes = nib.orientations.aff2axcodes(orig_affine)
```

Comment:
I feel confused by the "old_affine" and "orig_affine" here, maybe try to keep them consistent?

Thanks.